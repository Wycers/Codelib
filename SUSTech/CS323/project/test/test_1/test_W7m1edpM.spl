int add(int a, int b){
    return a+b;
}

int main(){
    int a, b, c;

    a = 3;
    b = 4;
    printf(add(a,b));

    return 0;
}