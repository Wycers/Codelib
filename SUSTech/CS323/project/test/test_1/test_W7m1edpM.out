Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: add
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: a
          COMMA
          VarList (1)
            ParamDec (1)
              Specifier (1)
                TYPE: int
              VarDec (1)
                ID: b
        RP
      CompSt (1)
        LC
        StmtList (2)
          Stmt (2)
            RETURN
            Exp (2)
              Exp (2)
                ID: a
              PLUS
              Exp (2)
                ID: b
            SEMI
        RC
    ExtDefList (5)
      ExtDef (5)
        Specifier (5)
          TYPE: int
        FunDec (5)
          ID: main
          LP
          RP
        CompSt (5)
          LC
          DefList (6)
            Def (6)
              Specifier (6)
                TYPE: int
              DecList (6)
                Dec (6)
                  VarDec (6)
                    ID: a
                COMMA
                DecList (6)
                  Dec (6)
                    VarDec (6)
                      ID: b
                  COMMA
                  DecList (6)
                    Dec (6)
                      VarDec (6)
                        ID: c
              SEMI
          StmtList (8)
            Stmt (8)
              Exp (8)
                Exp (8)
                  ID: a
                ASSIGN
                Exp (8)
                  INT: 3
              SEMI
            StmtList (9)
              Stmt (9)
                Exp (9)
                  Exp (9)
                    ID: b
                  ASSIGN
                  Exp (9)
                    INT: 4
                SEMI
              StmtList (10)
                Stmt (10)
                  Exp (10)
                    ID: printf
                    LP
                    Args (10)
                      Exp (10)
                        ID: add
                        LP
                        Args (10)
                          Exp (10)
                            ID: a
                          COMMA
                          Args (10)
                            Exp (10)
                              ID: b
                        RP
                    RP
                  SEMI
                StmtList (12)
                  Stmt (12)
                    RETURN
                    Exp (12)
                      INT: 0
                    SEMI
          RC
